lib/cdo/aws/resource_management/aws_resource_killer.rb

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env ruby
+require 'aws-sdk-ec2'
+require 'aws-sdk-cloudformation'
+require 'cdo/chat_client'
+
+class AwsResourceKiller
+  # Only exposed method. It can be executed multiple times.
+  # This library is meant to be used to delete resources and release resources.
+  def delete
+    log("Trying to FIND instances...")
+    instances_to_delete = find_instances_for_deletion
+    if instances_to_delete.empty?
+      log("nothing found for deletion... aborting")
+      return
+    end
+    log("This instances were found")
+    print_instances_ids(instances_to_delete)
+
+    log("Trying to TAG instances...")
+    tagged_instance_for_deletion = tag_instances_for_deletion(instances_to_delete)
+    if tagged_instance_for_deletion.empty?
+      log("nothing found for deletion... aborting")
+      return
+    end
+    log("This instances have tag for deletion")
+    print_instances_ids(tagged_instance_for_deletion)
+
+    log("Trying to DELETE instances...")
+    deleted_instances, exceptions = delete_tagged_instances(tagged_instance_for_deletion)
```

Comment:
If these instances are part of cloudformation stacks, we should delete the stack rather than simply deleting the instance.

Reply:
This is the abstract method.

In ec2_instance_killer. I implemented it to delete the stack

Reply:
I will read more :sweat_smile: